dot/digest/README.md

```diff
@@ -0,0 +1,95 @@
+[blockchain runtime](https://docs.substrate.io/v3/getting-started/glossary/#runtime). This package deals exclusively
+with the handling of [consensus digests](https://crates.parity.io/sp_runtime/enum.DigestItem.html#variant.Consensus),
+which are messages that originate from a blockchain runtime and are consumed by consensus engines. Consensus digests are
+formally described in
+[Section 5.1.2 of the Polkadot Specification](https://w3f.github.io/polkadot-spec/develop/_common_consensus_structures.html).
+What follows is a description of Gossamer's consensus digest `Handler`, including the consensus engines the `Handler`
+supports, and the messages the `Handler` may receive & corresponding actions it can dispatch.
+
+## Handler
+
+The [digest `Handler`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot/digest#Handler) implements the
+[`Service` interface](https://pkg.go.dev/github.com/ChainSafe/gossamer/lib/services#Service), which means that it
+exposes `Start` and `Stop` functions. When a
+[Gossamer `Node`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot#Node) is started, a digest `Handler` is created
+with the [`NewHandler`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot/digest#NewHandler) function and started
+along with the Gossamer node's other services. The digest `Handler` maintains two `goroutines`, `handleBlockImport` and
+`handleBlockFinalisation`, which handle GRANDPA changes that occur when blocks are imported and finalised,
+respectively - these `goroutines` listen for messages on channels provided by the
+[`BlockState`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot/state#BlockState).
+
+The digest `Handler` also exposes two public functions:
+[`NextGrandpaAuthorityChange`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot/digest#Handler.NextGrandpaAuthorityChange)
+and [`HandleDigests`](https://pkg.go.dev/github.com/ChainSafe/gossamer/dot/digest#Handler.HandleDigests).
+`NextGrandpaAuthorityChange` is consumed by the
+[GRANDPA service](https://pkg.go.dev/github.com/ChainSafe/gossamer/lib/grandpa#Service) and is used during the pre-vote
+and pre-commit [equivocation stages](https://medium.com/polkadot-network/polkadot-consensus-part-2-grandpa-fb1963ef6c70)
+to determine the block number of the next GRANDPA authority change.
+
+`HandleDigests` is invoked by the core service for each new block; this function reads the consensus digests from the
+block and dispatches actions based on the messages the digests contain. The digest `Handler` supports digests from the
+BABE (authorship) and GRANDPA (finalisation) consensus engines - the following sections describe the messages in these
+digests and the actions Gossamer takes when it receives them.
+
+## BABE Messages
+
+[BABE](https://wiki.polkadot.network/docs/learn-consensus#block-production-babe) is a proof-of-stake block authorship
+algorithm that randomly assigns block authorship rights over periods of time referred to as "epochs". One of the
```

Comment:
```suggestion
algorithm that assigns block authorship rights over periods of time referred to as "slots", which are a subdivision of an "epoch". One of the
```

BABE doesn't assign "randomly" exactly, it appears random, but it's actually not